src/SFA.DAS.EmployerFinance.MessageHandlers/CommandHandlers/DraftExpireFundsCommandHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using NServiceBus;
+using SFA.DAS.EmployerFinance.Data;
+using SFA.DAS.EmployerFinance.Interfaces;
+using SFA.DAS.EmployerFinance.Messages.Commands;
+
+namespace SFA.DAS.EmployerFinance.MessageHandlers.CommandHandlers
+{
+    public class DraftExpireFundsCommandHandler : IHandleMessages<DraftExpireFundsCommand>
+    {
+        private readonly ICurrentDateTime _currentDateTime;
+        private readonly IEmployerAccountRepository _accountRepository;
+
+        public DraftExpireFundsCommandHandler(ICurrentDateTime currentDateTime, IEmployerAccountRepository accountRepository)
+        {
+            _currentDateTime = currentDateTime;
+            _accountRepository = accountRepository;
+        }
+        public async Task Handle(DraftExpireFundsCommand message, IMessageHandlerContext context)
+        {
+            var now = _currentDateTime.Now;
+            var accounts = await _accountRepository.GetAllAccounts();
+            var commands = accounts.Select(a => new DraftExpireAccountFundsCommand { AccountId = a.Id, DateTo = message.DateTo});
+
+            var tasks = commands.Select(c =>
+            {
+                var sendOptions = new SendOptions();
+
+                sendOptions.RequireImmediateDispatch();
+                sendOptions.RouteToThisEndpoint();
+                sendOptions.SetMessageId($"{nameof(DraftExpireAccountFundsCommand)}-{now.Year}-{now.Month}-{c.AccountId}");
```

Comment:
Does setting the message id achieve anything?  Other than making it a pain in the arse to rerun?

Reply:
Removed